### PR TITLE
xorg: fix system packages

### LINKED ISSUE #5385: [package] xorg/system: xorg-x11-xkb-utils-devel is not available on fedora 34
`xorg-x11-xkb-utils-devel` does not exist on Fedora 34 anymore.
See https://src.fedoraproject.org/rpms/xorg-x11-xkb-utils/blob/rawhide/f/dead.package

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **xorg/system**
  * Operating System+version: **Fedora Linux 34**
  * Compiler+version: **GCC 11**
  * Conan version: **conan 1.36.0**
  * Python version: **Python 3.9.4**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler=gcc
compiler.version=11
compiler.libcxx=libstdc++11
build_type=Release
[options]
[build_requires]
[env]
```

### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
export CONAN_SYSREQUIRES_MODE=verify
conan install xorg/system
libxcb-devel-1.13.1-7.fc34.x86_64
Package already installed: libxcb-devel
libxcb-devel-1.13.1-7.fc34.x86_64
package libfontenc-devel is not installed
package libXaw-devel is not installed
package libXcomposite-devel is not installed
package libXcursor-devel is not installed
package libXdmcp-devel is not installed
package libXft-devel is not installed
package libXtst-devel is not installed
package libXinerama-devel is not installed
package xorg-x11-xkb-utils-devel is not installed
package libXrandr-devel is not installed
package libXres-devel is not installed
package libXScrnSaver-devel is not installed
package libXvMC-devel is not installed
package xorg-x11-xtrans-devel is not installed
package xcb-util-wm-devel is not installed
package xcb-util-image-devel is not installed
package xcb-util-keysyms-devel is not installed
package xcb-util-renderutil-devel is not installed
package libXdamage-devel is not installed
package libXxf86vm-devel is not installed
package libXv-devel is not installed
package xkeyboard-config-devel is not installed
package xcb-util-devel is not installed
Not updating system_requirements. CONAN_SYSREQUIRES_MODE=verify
Running: sudo dnf install -y libXScrnSaver-devel libXaw-devel libXcomposite-devel libXcursor-devel libXdamage-devel libXdmcp-devel libXft-devel libXinerama-devel libXrandr-devel libXres-devel libXtst-devel libXv-devel libXvMC-devel libXxf86vm-devel libfontenc-devel xcb-util-devel xcb-util-image-devel xcb-util-keysyms-devel xcb-util-renderutil-devel xcb-util-wm-devel xkeyboard-config-devel xorg-x11-xkb-utils-devel xorg-x11-xtrans-devel
Last metadata expiration check: 0:57:59 ago on vr 30 apr 2021 23:37:27.
No match for argument: xorg-x11-xkb-utils-devel
Error: Unable to find a match: xorg-x11-xkb-utils-devel
xorg/system: ERROR: while executing system_requirements(): Command 'sudo dnf install -y libXScrnSaver-devel libXaw-devel libXcomposite-devel libXcursor-devel libXdamage-devel libXdmcp-devel libXft-devel libXinerama-devel libXrandr-devel libXres-devel libXtst-devel libXv-devel libXvMC-devel libXxf86vm-devel libfontenc-devel xcb-util-devel xcb-util-image-devel xcb-util-keysyms-devel xcb-util-renderutil-devel xcb-util-wm-devel xkeyboard-config-devel xorg-x11-xkb-utils-devel xorg-x11-xtrans-devel' failed
ERROR: Error in system requirements
```

</details>

### PR DESCRIPTION
**Pull request by ericLemanissier**
fixes #5385

Specify library name and version:  **xorg/system**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xorg/all/conanfile.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def system_requirements(self):
             elif tools.os_info.with_yum or tools.os_info.with_dnf:
                 packages = ["libxcb-devel", "libfontenc-devel", "libXaw-devel", "libXcomposite-devel",
                             "libXcursor-devel", "libXdmcp-devel", "libXft-devel", "libXtst-devel", "libXinerama-devel",
-                            "xorg-x11-xkb-utils-devel", "libXrandr-devel", "libXres-devel", "libXScrnSaver-devel", "libXvMC-devel",
+                            "libxkbfile-devel", "libXrandr-devel", "libXres-devel", "libXScrnSaver-devel", "libXvMC-devel",
```
Comment (Croydon) [inline recipes/xorg/all/conanfile.py:57]: I think we should make a switch based on the OS version in order to support old and new versions

**Review by Croydon — CHANGES_REQUESTED** [review]

File: recipes/xorg/all/conanfile.py (hunk 1 above)
Comment (ericLemanissier) [reply recipes/xorg/all/conanfile.py:57]: it should be fine: libxkbfile-devel is a requirement of xorg-x11-xkb-utils-devel, which provides a single file: xkbcomp.pc, and xkbcomp is not referenced in https://github.com/conan-io/conan-center-index/blob/master/recipes/xorg/all/conanfile.py#L81. Let's wait for the next run on https://github.com/bincrafters/system-packages-checks/actions/workflows/cron.yml

Comment (ericLemanissier) [reply recipes/xorg/all/conanfile.py:57]: the only failure is on fedora 34, but it is unrelated: conan settings do not have gcc 11 yet : https://github.com/bincrafters/system-packages-checks/runs/2482530226?check_suite_focus=true#step:6:108

Comment (Croydon) [reply recipes/xorg/all/conanfile.py:57]: Thanks for the information!

**Review by Croydon — APPROVED** [review]

**Review by madebr — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**